### DISCUSSION TITLE
member_descriptor is not JSON serializable ERROR

### DISCUSSION THREAD

**Discussion by icekoid** [Rapptz/discord.py#5966] [Q&A]
My code:
```
import discord
from discord.ext import commands
import random

intents = discord.Intents.default()
intents.members = True

client = commands.Bot(command_prefix=".", intents = intents)


@client.command()
async def myNameIs(ctx):
    myName = discord.Message.content
    await ctx.author.edit(nick = myName)
    await ctx.send(f"Nickname was changed.")
```

It always gives me this error code and i dont know what it means and cant find info on the internet.

`discord.ext.commands.errors.CommandInvokeError: Command raised an exception: TypeError: Object of type member_descriptor is not JSON serializable`

**Comment by Rapptz** [comment]
This isn't how you access variables in Python, I recommend familiarizing yourself with [classes](https://docs.python.org/3/tutorial/classes.html#class-and-instance-variables).

`discord.Message` is a class, and `discord.Message.content` is a descriptor of that class. It doesn't refer to anything actually concrete. You're probably looking for `ctx.message.content`. However, the way you're doing this command makes it seem as if you want to do a parameter and change the nickname to what the user has passed. The command framework [documentation](https://discordpy.readthedocs.io/en/latest/ext/commands/commands.html) goes over what you want to do. It's a simple as writing:

```py
@client.command()
async def myNameIs(ctx, *, nickname: str):
    await ctx.author.edit(nick=nickname)
    await ctx.send('Nickname was changed.')
```

Please note that in order to use this library it is strongly recommended to be familiar with basic Python concepts -- not doing so will make things significantly more difficult than they should be.
